lib/dnsimple/client/registrar.rb

```diff
@@ -15,29 +35,32 @@ module Registrar
       # @raise  [RequestError] When the request fails.
       def register(account_id, domain_name, attributes = {}, options = {})
         Extra.validate_mandatory_attributes(attributes, [:registrant_id])
-        options  = options.merge(attributes)
-        response = client.post(Client.versioned("/%s/registrar/domains/%s/registration" % [account_id, domain_name]), options)
+        endpoint = Client.versioned("/%s/registrar/domains/%s/registration" % [account_id, domain_name])
+        response = client.post(endpoint, options.merge(attributes))
 
         Dnsimple::Response.new(response, Struct::Domain.new(response["data"]))
       end
 
-      # Checks whether a domain is available to be registered.
+      # Starts the transfer of a domain to DNSimple.
       #
-      # @see https://developer.dnsimple.com/v2/registrar/#check
+      # @see https://developer.dnsimple.com/v2/registrar/#transfer
       #
-      # @example Check whether example.com is available.
-      #   client.registrar.check(1010, "example.com")
+      # @example Initiate the transfer for example.com using the contact 1234 as registrant:
+      #   client.registrar.transfer(1010, "example.com", registrant_id: 1234, auth_info: "x1y2z3")
       #
       # @param  [Fixnum] account_id the account ID
-      # @param  [#to_s] domain_name The domain name to check.
+      # @param  [#to_s] domain_name The domain name to transfer.
+      # @param  [Hash] attributes
       # @param  [Hash] options
-      # @return [Struct::DomainCheck]
+      # @return [Struct::Domain]
       #
       # @raise  [RequestError] When the request fails.
-      def check(account_id, domain_name, options = {})
-        response = client.get(Client.versioned("/%s/registrar/domains/%s/check" % [account_id, domain_name]), options)
+      def transfer(account_id, domain_name, attributes = {}, options = {})
+        Extra.validate_mandatory_attributes(attributes, [:registrant_id, :auth_info])
```

Comment:
`auth_info` is not required for all the TLDs. It depends on the TLD (e.g. `UK` don't use the `auth_info`). We can't validate it at client level, we have to send the request and delay the delegation to the app.